src/DockAreaTitleBar.cpp

```diff
@@ -137,6 +152,13 @@ DockAreaTitleBarPrivate::DockAreaTitleBarPrivate(CDockAreaTitleBar* _public) :
 
 }
 
+DockAreaTitleBarPrivate::~DockAreaTitleBarPrivate()
+{
+    delete TabsMenuButton;
+    delete UndockButton;
+    delete CloseButton;
```

Comment:
No need to do this, these are assigned a parent when added to the layout. Qt will use parent destruction order to destroy it afterwards.

Reply:
@roliav, unfortunately, we cannot rely on it- see my https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/pull/103 - buttons were left after app exit if DockAreaHasCloseButton was set to 'false' (see DockAreaTitleBarPrivate::createButtons() code)

Reply:
Sorry @mvidelgauz, I completely missunderstood the PR. You are right.